build.gradle

```diff
@@ -423,42 +483,11 @@ subprojects {
       fatJarConfig
 
       testOutput {
-        extendsFrom testRuntime
+        extendsFrom testRuntimeClasspath
         transitive = true
       }
     }
 
-    sourceSets {
-      main {
-        java {
-          compileClasspath += configurations.compileOnly
-        }
-      }
-      test {
-        java {
-          compileClasspath += configurations.compileOnly
-          runtimeClasspath += configurations.compileOnly
-        }
-      }
-    }
```

Comment:
This bit is partly redundant since `compileClasspath` already `extendsFrom compileOnly`.

Adding the `compileClasspath` to `test{Compile,Runtime}Classpath` is achieved on line 296:

```groovy
    // it's rather convenient to allow compile and runtime classpaths for tests inherit from the main compile classpath
    testImplementation.extendsFrom compileClasspath
```